test/devices_list_test.rb

```diff
@@ -58,6 +58,14 @@
         )
       end
 
+      it "filters by nil values" do
```

Comment:
It makes more sense if you read the spec as a whole (with all the variants of "it filters by xxx"). Anyway, it will be better like this?

```it "filters by missing values" do```

or maybe

```it "filters checking for nil attributes"```

or something in between?